repair/repair.hh

```diff
@@ -67,16 +67,28 @@ struct repair_uniq_id {
 };
 std::ostream& operator<<(std::ostream& os, const repair_uniq_id& x);
 
-struct node_ops_info {
+class node_ops_info {
+public:
     utils::UUID ops_uuid;
     shared_ptr<abort_source> as;
     std::list<gms::inet_address> ignore_nodes;
 
+private:
+    optimized_optional<abort_source::subscription> _abort_subscription;
+    sharded<abort_source> _sas;
```

Comment:
I dont understand why this is sharded if the wrapping class is alreasy too.

Reply:
this is sharded while the wrapping class is shared.
god is in the details :)

Reply:
The start() and stop() methods confused me into thinking `node_ops_info` is `sharded<>` too.
I think these are weird sounding methods to have on an `info` class, but its fine. You mentioned this code is subject to further changes, maybe we can revisit this code after that and see if it needs some shaking-up.

Reply:
Yes.  I actually tried a different approach that would be closer to where we want to get in https://github.com/bhalevy/scylla/commit/eed170f35e7ea6b3d1d095bf448bb582d7f8845a
but I decided that the churn is too high for now.

Reply:
Ok, I'll queue once CI approves.